common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/DockerToActivationFileLogStore.scala

```diff
@@ -123,7 +123,11 @@ class DockerToActivationFileLogStore(system: ActorSystem, destinationDirectory:
                            container: Container,
                            action: ExecutableWhiskAction): Future[ActivationLogs] = {
 
-    val logs = container.logs(action.limits.logs.asMegaBytes, action.exec.sentinelledLogs)(transid)
+    // wait for a sentinel only if no container (developer) error occurred to avoid
+    // that log collection continues if the action code still logs after timeout
+    val sentinel = action.exec.sentinelledLogs && !activation.response.isContainerError
```

Comment:
Two questions:

1. The `ActivationResponse` is a container error if the response's status code is an `ActivationResponse.DeveloperError`. This status code is set if the action init / run times out. Your change is desired in case of a timeout.
   In addition, a developer error is created for other failures as well. Did you check all code locations where a developer error is created to make sure you do not introduce undesired behaviour for other failures?

2. You are preventing on-going log file reading for developer errors. Did you check whether it could be helpful to change log file reading behaviour for other errors as well?

Reply:
We have changed the implementation to use an activation annotation `timeout` (boolean) to indicate action timeouts as proposed by @sven-lange-last.